HIP/hip-583.md

```diff
@@ -10,220 +10,452 @@ status: Accepted
+4. If the `Key` is an `ECDSA_secp256k1` key:
+    1. If the alias is already in use, we use that account and do not create a new one
+    2. If the derived `evm address` from the key is already in use, we use that account and do not create a new one
+    3. If the alias is not in use, we create a new account and use the given key as the key on the account.
 
-## Backwards Compatibility
+If we do create a new account from either `CryptoCreate` or `CryptoTransfer`, and the `alias` is an `ECDSA_secp256k1`
+key, then we reserve in the aliases for **both** the raw protobuf-encoded `Key` bytes **and** the derived evm address.
+Essentially, this allows the account created with an `ECDSA_secp256k1` to be addressed by that alias, the
+`evm address alias`, and the `account-num alias`.
 
-Existing accounts are unaffected by the expansion and the additional alias field on the transaction protobuf is already supported.
+##### `evm address alias`
+
+When an attempt is made to transfer value to an account with an alias with `CryptoTransfer`, we check whether the alias
+has exactly 20 bytes. Then after making sure it is not an `account-num alias`, we assume the alias is an
+`evm address alias` and:
+
+1. If there is an account with this `evm address alias` (whether because the `alias` on the `Account` is this
+   `evm address alias`, or because it is a `key alias` of type `ECDSA_secp256k1` that, when derived, maps to
+   this `evm address alias`), then we use that account and do not create a new one.
+2. Otherwise, we create a `Hollow account` and set the `alias` to be the `evm address alias`
+
+Since we reserve the `evm address alias` and since an `ECDSA_secp256k1` `key alias` checks for collisions with the
+`evm address alias`, we effectively reserve the `key alias` with the corresponding `ECDSA_secp256k1` key slot as well.
+This prevents collisions between compatible `key alias` and `evm address alias`.
+
+Backwards Compatibility
+------------------------------------------------------------------------------------------------------------------------
+
+Existing accounts are unaffected by the expansion and the additional alias field on the transaction protobuf is already
+supported.
 
 `CryptoTransfer` auto-account creation is un-impacted and will still support the provision of the public key. 
 
-### Mirror Node Impications
+### Mirror Node Implications
 
-The Mirror Node can already ingest account creation transactions with alias values.
-This will only increate the applicable parent trnasactions that can cause this flow.
-Where it was limited to CryptoTransfer it will now include CryptoCreate
+The Mirror Node can already ingest account creation transactions with alias values. This will only increase the
+applicable parent transactions that can cause this flow. Where it was limited to CryptoTransfer it will now include
+CryptoCreate, and in both cases it will now support `evm address` formatted aliases.
```

Comment:
We need to verify that the mirror node has the ability to find info by evm address alias in all cases (I think it does, but we need to know).

There is some additional complexity if you are processing on raw transactions. Suppose you are processing all transactions, and filtering them if they apply to the accounts you care about. Some transactions may be based on `shard.realm.num`, and some may be based on alias. Such programs have to know what real account number is associated with each alias.

Also, the mirror node needs to be prepared for hollow accounts. If a thousand transactions have been made by sending value to a hollow account, and then it is made "real", then queries on the mirror node using the `shard.realm.num` must return all the previous transactions as well.

Reply:
It would be good to have an API on mirror node where, if I give any one of my alias forms, it will return all alias forms that are supported. For example, suppose I have account `0.0.10000` and it has a `key alias` that is an ECDSA key. If I query that mirror node API, I should see `0.0.10000`, the long-zero alias form, the `key alias` form, and the `evm address alias` form associated with that `key alias`! This can be a special query.